### PR TITLE
shivam-51: Add log file paths to dapr list 1228

### PR DESCRIPTION
**Pull request by shivam-51**
# Description

Add paths of app and dapr logs to `dapr list` command.

## Issue reference

Please reference the issue this PR will close: #1228 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Comment by mukundansundar** [comment]
@shivam-51 Can you please add E2E for this change?

**Comment by pravinpushkar** [comment]
Multi app run template introduces `log destination` fields for daprd and app. We have to take into account if the log destination is set as `console` then what values it captures.

**Comment by shivam-51** [comment]
> Multi app run template introduces `log destination` fields for daprd and app. We have to take into account if the log destination is set as `console` then what values it captures. @shivam-51

@pravinpushkar Can you share some documentation for this property? I am not able to understand it's significance.
I checked https://docs.dapr.io/developing-applications/local-development/multi-app-dapr-run/multi-app-template/

**Comment by shivam-51** [comment]
> Multi app run template introduces `log destination` fields for daprd and app. We have to take into account if the log destination is set as `console` then what values it captures. @shivam-51

- For `appLogDestination: console` show `appLogPath: console`
- For `appLogDestination: file` show `appLogPath: <file path>`
- For `appLogDestination: fileAndConsole` show `appLogPath: <file path>` giving file more priority than console

@pravinpushkar @mukundansundar What do you think?

**Comment by philliphoff** [comment]
@shivam-51 @pravinpushkar @mukundansundar My vote would be that, if the log is written to a file, then add/populate the `appLogDestination` metadata property.  If the log is not (e.g. being solely written to the console), then simply omit that metadata property (or set to `null` or the empty string `""`).  I would rather not have "special strings" that tooling needs to distinguish from otherwise valid paths.

**Comment by pravinpushkar** [comment]
> @shivam-51 @pravinpushkar @mukundansundar My vote would be that, if the log is written to a file, then add/populate the `appLogDestination` metadata property. If the log is not (e.g. being solely written to the console), then simply omit that metadata property (or set to `null` or the empty string `""`). I would rather not have "special strings" that tooling needs to distinguish from otherwise valid paths.

+1 to update log paths only when written to only files. In other cases the logs will anyway be available to the console and tolling can use that. 

@philliphoff However one question -  if there are multiple apps writing logs to console then how will the tooling distinguish which log written by which app ?

**Comment by philliphoff** [comment]
>However one question - if there are multiple apps writing logs to console then how will the tooling distinguish which log written by which app ?

@pravinpushkar If the application output is only going to the console, then it'd be unavailable to tooling anyway (unless it was the one to have started the Dapr process and captured the `stdout` itself).  If no path metadata is present, then tooling would simply assume it's being captured someplace else and not offer the user the option to view the logs.

**Comment by shivam-51** [comment]
> Can you please add E2E for this change? Please see #1200 for reference.

@mukundansundar I have identified the tests need to go into `list_test.go` but I am not able to test locally. Most of the tests fail somehow.

**Comment by mukundansundar** [comment]
> > Can you please add E2E for this change? Please see #1200 for reference.
> 
> @mukundansundar I have identified the tests need to go into `list_test.go` but I am not able to test locally. Most of the tests fail somehow.

Could you explain what errors that you are seeing?

File: cmd/run.go [hunk 1]
```diff
@@ -540,6 +540,16 @@ func executeRun(runTemplateName, runFilePath string, apps []runfileconfig.App) (
 		// Update extended metadata with run file path.
 		putRunTemplateNameInMeta(runState, runTemplateName)
 
+		// Update extended metadata with app log file path.
+		if app.AppLogDestination != "console" {
```
Comment (mukundansundar) [inline cmd/run.go:544]: I think there is a constant for `console`, please use that.

**Review by mukundansundar — CHANGES_REQUESTED** [review]

**Comment by shivam-51** [comment]
> Could you explain what errors that you are seeing?

That's okay. I ran the tests on my fork using github actions.

File: cmd/run.go (hunk 1 above)
Comment (shivam-51) [reply cmd/run.go]: Done.

**Review by mukundansundar — CHANGES_REQUESTED** [review]
@shivam-51 Almost there, thanks for adding the negative scenarios. Can you add the positive test case as part of the run file scenarios where the list is actually populated with the file path? Inside this function here https://github.com/dapr/cli/blob/master/tests/e2e/standalone/stop_with_run_template_test.go#L121

**Comment by shivam-51** [comment]
@mukundansundar can you have a look please?

**Review by mukundansundar — APPROVED** [review]

**Pull request merged**